src/controllers/auth.js

```diff
@@ -155,4 +163,48 @@ async function logout (req, res) {
   })
 }
 
-export { init, confirmed, logout }
+/**
+ * Controller that initiates the premium upgrade flow.
+ *
+ * Redirects to subplat. If the user is already subscribed, they are redirected
+ * back to the dashboard.
+ *
+ * @param {object} req The express request object
+ * @param {object} res The express response object
```

Comment:
Just FYI right now the TypeScript compiler doesn't run over all the code, so you'd need to add these files to `tsconfig.json` in the root to opt them in. I tried doing it just now and you'd have to fix a lot of code unrelated to this PR :) but the good news is that I don't see any serious type safety issues in this PR.

However, adding the proper types in JSDoc is always good, and once we enable `tsc` globally it'll be checked.

Reply:
Agree with the overall idea, but since it looks like everywhere else in the code uses the plain object I think it wouldn't be the right place to do it in this PR. Sounds like it's in the plan to do this for monitor in the future though (right?) which is great.

Reply:
We're adopting it incrementally rather than trying to annotate the entire codebase in one go, so some of the newer files ([example](https://github.com/mozilla/blurts-server/blob/0ea1ef5023e4621edc3870e921882daf20a01f2f/src/controllers/exposureScan.js#L10)) have proper type annotations, whereas some of the older ones have none, incomplete or inaccurate annotations. So yeah, boy-scout-rule-wise it's nice to add them to leave things cleaner than you found them, but not having them is not a blocker.

Reply:
Just cause you got me curious on a Thursday evening:
```sh
git grep -En "@param .* (req|res) " src | cat

src/controllers/auth.js:148: * @param {object} req Contains session.user
src/controllers/auth.js:149: * @param {object} res Redirects to homepage
src/controllers/auth.js:171: * @param {object} req The express request object
src/controllers/auth.js:172: * @param {object} res The express response object
src/controllers/auth.js:191: * @param {object} req The express request object
src/controllers/auth.js:192: * @param {object} res The express response object
src/controllers/breaches.js:58: * @param {object} req containing {user, body: {affectedEmail, breachId, resolutionsChecked}}
src/controllers/breaches.js:63: * @param {object} res JSON object containing the updated breach resolution
src/middleware/error.js:21: * @param {object} req request object
src/middleware/error.js:22: * @param {object} res response object
src/utils/email.js:183: * @param {object} req Request should contain a `token` and `hash` query param.
src/utils/email.js:199: * @param {object} req Request that should contain a `token` query param
```